### PR TITLE
First draft of Consensus gRPC interface

### PR DESCRIPTION
**Pull request by canepat**
First draft of Consensus gRPC interface

### CODE REVIEW

File: consensus_engine/consensus.proto [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+message FinalizeResponse {
+    Result result = 1;
+    types.H256 miner_reward = 2;
+    repeated types.H256 uncle_rewards = 3;
+}
+
+service ConsensusEngine {
+    rpc GetAuthor(GetAuthorRequest) returns(GetAuthorResponse);
+    rpc VerifyHeader(stream VerifyHeaderRequest) returns(stream VerifyHeaderResponse);
+    rpc VerifyHeaders(stream VerifyHeadersRequest) returns(stream VerifyHeadersResponse);
```
Comment (JekaMas) [inline consensus_engine/consensus.proto:115]: Do we need both VerifyHeader and VerifyHeaders?

File: consensus_engine/consensus.proto [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+message FinalizeResponse {
+    Result result = 1;
+    types.H256 miner_reward = 2;
+    repeated types.H256 uncle_rewards = 3;
+}
+
+service ConsensusEngine {
+    rpc GetAuthor(GetAuthorRequest) returns(GetAuthorResponse);
+    rpc VerifyHeader(stream VerifyHeaderRequest) returns(stream VerifyHeaderResponse);
```
Comment (JekaMas) [inline consensus_engine/consensus.proto:114]: Is this api only for Ethash case?

File: consensus_engine/consensus.proto (hunk 1 above)
Comment (canepat) [reply consensus_engine/consensus.proto:115]: Good catch. I've just started porting the [Engine](https://github.com/ledgerwatch/turbo-geth/blob/master/consensus/consensus.go#L59) Go interface to gRPC proto format, but I agree both are not needed: only `VerifyHeaders` is sufficient, if needed `headers` field can contain just one block header.

File: consensus_engine/consensus.proto (hunk 2 above)
Comment (canepat) [reply consensus_engine/consensus.proto:114]: It should be the generic interface between TG core and Consensus Engine module, the latter being able to implement any of the existing consensus algorithms.

File: consensus_engine/consensus.proto (hunk 1 above)
Comment (JekaMas) [reply consensus_engine/consensus.proto:115]: I see. The work on extracting consensus interface is in another branch. At the moment we have an interface for VerifyHeader that doesn't read from DB directly:
https://github.com/ledgerwatch/turbo-geth/blob/3af902abeafd7ca0fd9e8b06e15e64fe7d09ec86/consensus/consensus.go#L103
https://github.com/ledgerwatch/turbo-geth/blob/3af902abeafd7ca0fd9e8b06e15e64fe7d09ec86/consensus/engine.go#L47

We can discuss API and current problems.

**Review by AlexeyAkhunov — APPROVED** [review]

**Pull request merged**